chain/chain/src/flat_storage_resharder.rs

```diff
@@ -0,0 +1,943 @@
+        )?,
+        col::ACCESS_KEY => copy_kv_to_child(
+            &status,
+            key,
+            value,
+            store_update,
+            parse_account_id_from_access_key_key,
+        )?,
+        col::RECEIVED_DATA => copy_kv_to_child(
+            &status,
+            key,
+            value,
+            store_update,
+            parse_account_id_from_received_data_key,
+        )?,
+        col::POSTPONED_RECEIPT_ID | col::PENDING_DATA_COUNT | col::POSTPONED_RECEIPT => {
+            copy_kv_to_child(&status, key, value, store_update, |raw_key: &[u8]| {
+                parse_account_id_from_trie_key_with_separator(
+                    key_column_prefix,
+                    raw_key,
+                    ALL_COLUMNS_WITH_NAMES[key_column_prefix as usize].1,
+                )
+            })?
+        }
+        col::DELAYED_RECEIPT_OR_INDICES
+        | col::PROMISE_YIELD_INDICES
+        | col::PROMISE_YIELD_TIMEOUT
+        | col::PROMISE_YIELD_RECEIPT
+        | col::BUFFERED_RECEIPT_INDICES
+        | col::BUFFERED_RECEIPT => {
+            // TODO(trisfald): implement logic and remove error log
+            let col_name = ALL_COLUMNS_WITH_NAMES[key_column_prefix as usize].1;
+            error!(target: "resharding", "flat storage resharding of {col_name} is not implemented yet!");
+        }
+        _ => unreachable!(),
+    }
+    Ok(())
+}
+
+/// Performs post-processing of shard splitting after all key-values have been moved from parent to children.
+/// `success` indicates whether or not the previous phase was successful.
+fn split_shard_task_postprocessing(
+    resharder: FlatStorageResharder,
+    task_status: FlatStorageReshardingTaskStatus,
+) {
+    let (parent_shard, split_status) = get_parent_shard_and_status(&resharder);
+    let SplittingParentStatus { left_child_shard, right_child_shard, flat_head, .. } = split_status;
+    let flat_store = resharder.runtime.store().flat_store();
+    info!(target: "resharding", ?parent_shard, ?task_status, ?split_status, "flat storage shard split task: post-processing");
+
+    let mut store_update = flat_store.store_update();
+    match task_status {
+        FlatStorageReshardingTaskStatus::Successful => {
+            // Split shard completed successfully.
+            // Parent flat storage can be deleted from the FlatStoreManager.
+            resharder
+                .runtime
+                .get_flat_storage_manager()
+                .remove_flat_storage_for_shard(parent_shard, &mut store_update)
+                .unwrap();
+            store_update.remove_flat_storage(parent_shard);
+            // Children must perform catchup.
+            for child_shard in [left_child_shard, right_child_shard] {
+                store_update.set_flat_storage_status(
+                    child_shard,
+                    FlatStorageStatus::Resharding(FlatStorageReshardingStatus::CatchingUp(
+                        flat_head.hash,
```

Comment:
I think this should be the "resharding hash" - the last block of the old shard layout.

Reply:
Will be fixed in another PR